### PR TITLE
Improve grammar, spelling and structure of HTTP-Header Format and processing model

### PR DESCRIPTION
**Pull request by danielkhan**
This PR contains a set of changes to the HTTP Header Format specification and the processing model document.
Except replacing the 512byte limit with the 5 list item limit, there should be no semantic changes included.

It contains
- Spelling fixes
- Grammatical fixes
- Changes that add more consistency in our wording
- Some restructuring to make the overall flow more coherent

This PR is meant to be a base for further discussions and should not be considered as final. Please review thoroughly, if the changes make sense and that no semantic changes gave slipped in.

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
@danielkhan I had feedback for original PR and now Alois is splitting it to multiple small ones. Have you created this PR for Alois to accept or for others to review? It's kind of hard to review as the comment may be related to the original text =)

**Comment by danielkhan** [comment]
@SergeyKanzhelev my PR builds on the CR branch that Alois created so eventually it should be merged back to his‘ and then ideally become part of the final version. I‘d appreciate it if you could look through it and add your thoughts because much of the original version was done by Adrian and you.

File: spec/20-HTTP_HEADER_FORMAT.md [hunk 1]
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
...
+Represents the ID of the whole trace forest as 16-byte array and is used to
+uniquely identify a distributed trace throughout all participating systems.
 
-`Trace-id` is used to uniquely identify a distributed trace. So implementation
-should generate globally unique values. Many algorithms of unique identification
-generation are based on some constant part - time or host based and a random
-value. There are systems that make random sampling decisions based on the value
-of `trace-id`. So to increase interoperability it is recommended to keep the
-random part on the right side of `trace-id` value.
+**Example:** `4bf92f3577b34da6a3ce929d0e0e4736`.
```
Comment (AloisReitbauer) [inline spec/20-HTTP_HEADER_FORMAT.md:103]: use uppercase for hexdig

File: spec/20-HTTP_HEADER_FORMAT.md [hunk 2]
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
...
 ```
 
-### Trace-id
+#### trace-id
 
-Is the ID of the whole trace forest. It is represented as a 16-bytes array, for
-example, `4bf92f3577b34da6a3ce929d0e0e4736`. All bytes `0` are considered
-invalid.
+Represents the ID of the whole trace forest as 16-byte array and is used to
+uniquely identify a distributed trace throughout all participating systems.
```
Comment (AloisReitbauer) [inline spec/20-HTTP_HEADER_FORMAT.md]: add detailed semantics

File: spec/20-HTTP_HEADER_FORMAT.md [hunk 3]
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
...
-When a system operates with a shorter `trace-id` - it is recommended to fill-in
-the extra bytes with random values rather than zeroes. Let's say the system
-works with a 8-byte `trace-id` like `3ce929d0e0e4736`. Instead of setting
-`trace-id` value to `0000000000000003ce929d0e0e4736` it is recommended to
-generate a value like `4bf92f3577b34da6a3ce929d0e0e4736` where
-`4bf92f3577b34da6a` is a random value or a function of time & host value. Note,
-even though a system may operate with a shorter `trace-id` for distributed trace
-reporting - full `trace-id` should be propagated to conform to the
-specification.
+> **Note:** Many algorithms of unique ID generation are based on some constant
```
Comment (AloisReitbauer) [inline spec/20-HTTP_HEADER_FORMAT.md]: for

File: spec/20-HTTP_HEADER_FORMAT.md [hunk 4]
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
...
+> part (time or host-based) and a random value. Some systems make random
```
Comment (AloisReitbauer) [inline spec/20-HTTP_HEADER_FORMAT.md:106]: Isn't this a too opinionated way to talk about randomness?

File: spec/20-HTTP_HEADER_FORMAT.md [hunk 5]
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
...
 
-Implementations HAVE TO ignore the `traceparent` when the `trace-id` is invalid.
-For instance, if it contains non-allowed characters.
+##### Rules
 
-### Parent-id
+* A `trace-id` with all bytes set to `0` is not allowed.
+* Implementations SHOULD generate globally unique values.
+* Implementations HAVE TO ignore the `traceparent` if the `trace-id` does not
+* comply with this specification.
```
Comment (danielkhan) [inline spec/20-HTTP_HEADER_FORMAT.md]: fix bullets

File: spec/20-HTTP_HEADER_FORMAT.md [hunk 6]
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
...
-An [8-bit field](https://en.wikipedia.org/wiki/Bit_field) that controls tracing
-flags such as sampling, trace level etc. These flags are recommendations given
-by the caller rather than strict rules to follow for three reasons:
+**Example:** `00f067aa0ba902b7`.
 
-1. Trust and abuse
-2. Bug in caller
-3. Different load between caller service and callee service might force callee
-   to down sample.
+**Note:** `parent-id` is sometimes also referred to as `span-id` as some
```
Comment (AloisReitbauer) [inline spec/20-HTTP_HEADER_FORMAT.md:135]: remove

File: spec/20-HTTP_HEADER_FORMAT.md [hunk 7]
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
...
 base16(version) = 00
 base16(trace-id) = 4bf92f3577b34da6a3ce929d0e0e4736
 base16(parent-id) = 00f067aa0ba902b7
 base16(trace-flags) = 00  // not recorded
 ```
 
-## Versioning of `traceparent`
+### Versioning of `traceparent`
+
+> **FIXME - what does this mean?** Implementation is opinionated about future version of the specification.
```
Comment (AloisReitbauer) [inline spec/20-HTTP_HEADER_FORMAT.md]: remove sentence

File: spec/20-HTTP_HEADER_FORMAT.md [hunk 8]
```diff
@@ -360,70 +362,94 @@ nblk-chr = %x21-2B / %x2D-3C / %x3E-7E
...
+  incoming requests SHOULD be combined in a single header according to the
+  [RFC7230](https://tools.ietf.org/html/rfc7230#page-24) and sent as a single
+  header in outgoing request.
+* Empty and whitespace-only `list-member` entries are allowed.
+* A participating system MUST propagate AT LEAST five `list-member` entries. It
+  MAY drop `list-member` entries that exceed this limit. When dropping entries,
+  the system SHOULD start dropping items starting at the right side (end) of the
+  list. When propagating `tracestate` with the excessive length, the assumption
+  SHOULD be that the receiver will trim the list to the length of five.
+* Libraries and platforms MUST accept empty `tracestate` headers, but SHOULD
```
Comment (AloisReitbauer) [inline spec/20-HTTP_HEADER_FORMAT.md:380]: TMI

File: spec/20-HTTP_HEADER_FORMAT.md [hunk 9]
```diff
@@ -467,13 +493,13 @@ the beginning of the list.
 Here is the list of allowed mutations:
```
Comment (AloisReitbauer) [inline spec/20-HTTP_HEADER_FORMAT.md:493]: wording

File: spec/20-HTTP_HEADER_FORMAT.md [hunk 10]
```diff
@@ -1,30 +1,41 @@
...
 If the receiving server is traced in the `rojo` tracing system, it carries over
-the state it received and adds a new entry with the position in its trace.
+the state it received and adds a new entry to the top of `tracestate`.
 
 ``` http
 traceparent: 00-0af7651916cd43dd8448eb211c80319c-00f067aa0ba902b7-01
 tracestate: rojo=00-0af7651916cd43dd8448eb211c80319c-00f067aa0ba902b7-01,congo=BleGNlZWRzIHRohbCBwbGVhc3VyZS4
 ```
 
+> **FIXME - the generic tracing system is defined in the rationale doc which is
```
Comment (danielkhan) [inline spec/20-HTTP_HEADER_FORMAT.md:37]: Clarify, if this information isn't too much information in this context

**Review by AloisReitbauer — APPROVED** [review]

**Pull request merged**